pkg/controllers/provisioningv2/rke2/rkecontrolplane/controlplane.go

```diff
@@ -52,3 +68,55 @@ func (h *handler) OnChange(obj *rkev1.RKEControlPlane, status rkev1.RKEControlPl
 	status.Initialized = Ready.IsTrue(cluster)
 	return status, nil
 }
+
+func (h *handler) updateControlPlaneStatus(cp *rkev1.RKEControlPlane, status rkev1.RKEControlPlaneStatus, previousErr error) (*rkev1.RKEControlPlane, error) {
```

Comment:
I'm not really sure this needs to be its own function, can we not just move this logic to the `OnRemove` below?